### ISSUE TITLE
error in process sentinel: Multibyte text in HTTP request: POST /v1/chat/completions HTTP/1.1...

### ISSUE THREAD

**Issue by JonatanSahar** [karthink/gptel#32]
Hi, 
It's been working great for me lately, but I just got the error below.
I've fed it an abstract and asked for a summary. Maybe it was too long?

Thanks!
```
{"model":"gpt-3.5-turbo","messages":[{"role":"system","content":"You are a large language model living in Emacs and a helpful assistant. Respond concisely."},{"role":"user","content":"summarize the following abstract: Sounds can arise from the environment and also predictably from many of our own movements, such as vocalizing, walking, or playing music. The capacity to anticipate these movement-related (reafferent) sounds and distinguish them from environmental sounds is essential for normal hearing1,2, but the neural circuits that learn to anticipate the often arbitrary and changeable sounds that result from our movements remain largely unknown. Here we developed an acoustic virtual reality (aVR) system in which a mouse learned to associate a novel sound with its locomotor movements, allowing us to identify the neural circuit mechanisms that learn to suppress reafferent sounds and to probe the behavioural consequences of this predictable sensorimotor experience. We found that aVR experience gradually and selectively suppressed auditory cortical responses to the reafferent frequency, in part by strengthening motor cortical activation of auditory cortical inhibitory neurons that respond to the reafferent tone. This plasticity is behaviourally adaptive, as aVR-experienced mice showed an enhanced ability to detect non-reafferent tones during movement. Together, these findings describe a dynamic sensory filter that involves motor cortical inputs to the auditory cortex that can be shaped by experience to selectively suppress the predictable acoustic consequences of movement."},{"role":"assistant","content":"The study explores the neural mechanisms that allow us to distinguish sounds that arise from our movements from those that come from the environment. They used mice in an acoustic virtual reality setting to identify the neural circuits responsible for suppressing movement-related sounds and the behavioral consequences of this experience. The findings suggest that motor cortical inputs to the auditory cortex shape a sensory filter that can suppress predictable acoustic consequences of movement and enhance auditory discrimination during movement."},{"role":"user","content":"what is this paper about? Sensory regions of the brain integrate environmental cues with copies of motor-related signals important for imminent and ongoing movements. In mammals, signals propagating from the motor cortex to the auditory cortex are thought to have a critical role in normal hearing and behaviour, yet the synaptic and circuit mechanisms by which these motor-related signals influence auditory cortical activity remain poorly understood. Using in vivo intracellular recordings in behaving mice, we find that excitatory neurons in the auditory cortex are suppressed before and during movement, owing in part to increased activity of local parvalbumin-positive interneurons. Electrophysiology and optogenetic gain- and loss-of-function experiments reveal that motor-related changes in auditory cortical dynamics are driven by a subset of neurons in the secondary motor cortex that innervate the auditory cortex and are active during movement. These findings provide a synaptic and circuit basis for the motor-related corollary discharge hypothesized to facilitate hearing and auditory-guided behaviours."},{"role":"assistant","content":"This paper investigates how motor-related signals from the motor cortex influence auditory cortical activity in mice, and the synaptic and circuit mechanisms underlying this interplay. The research findings suggest that a subset of neurons in the secondary motor cortex that innervate the auditory cortex and are active during movement drive motor-related changes in auditory cortical dynamics, likely facilitating hearing and auditory-guided behaviors. The study used in vivo intracellular recordings and optogenetics in mice."},{"role":"user","content":"what is this paper about? In the present study we investigated the neural code of sensory predictions. Grounded on a variety of empirical findings, we set out from the proposal that sensory predictions are coded via the top-down modulation of the sensory units whose response properties match the specific characteristics of the predicted stimulus (Albright, 2012; Arnal and Giraud, 2012). From this proposal, we derive the hypothesis that when the specific physical characteristics of the predicted stimulus cannot be advanced, the sensory system should not be able to formulate such predictions, as it would lack the means to represent them. In different conditions, participant   s self-paced button presses predicted either only the precise time when a random sound would be presented (random sound condition) or both the timing and the"}],"temperature":1.0}
error in process sentinel: Multibyte text in HTTP request: POST /v1/chat/completions HTTP/1.1

MIME-Version: 1.0

Connection: keep-alive

Extension: Security/Digest Security/SSL

Host: api.openai.com

Accept-encoding: gzip

Accept: */*

User-Agent: URL/Emacs Emacs/28.1 (Windows-NT; 32bit; x86_64-w64-mingw32)

Content-Type: application/json

Authorization: Bearer sk-c---------------------------------------------pS

Content-length: 4777

```

**Comment by karthink** [comment]
> Maybe it was too long?

That should produce a readable error from ChatGPT. This looks like another issue with curl or url-retrieve. Is `gptel-use-curl` set to `t` for you right now?

**Comment by jpe90** [comment]
@JonatanSahar Just a heads up- you should probably [rotate that API key](https://platform.openai.com/account/api-keys) if you haven't already.

**Comment by karthink** [comment]
I'm closing this now since gptel's code has changed significantly in the past few months. The bug in the `url-retrieve` method has been fixed in #4, and the Curl method has been rewritten. Feel free to re-open if this bug persists.
